### PR TITLE
Add Dactyl Manuform board (variants)

### PR DESCRIPTION
**Pull request by dunk2k**
Add variants:
- 4x5
- 4x6
- 5x6
- 5x7
- 6x6
- 6x7

### CODE REVIEW

File: boards/dactyl_manuform/5x6/main.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        KC.TRNS, KC.NO, KC.VOLU, KC.PGUP,   KC.NO, KC.NO,                         KC.NLCK, KC.P7, KC.P8,   KC.P9, KC.PMNS,   KC.NO,
+        KC.TRNS, KC.NO, KC.MUTE, KC.SLCK, KC.MPLY, KC.NO,                         KC.PAST, KC.P4, KC.P5,   KC.P6, KC.PPLS, KC.TRNS,
+        KC.TRNS, KC.NO, KC.VOLD, KC.PGDN,   KC.NO, KC.NO,                         KC.PSLS, KC.P1,  C.P2,   KC.P3, KC.PENT, KC.TRNS,
+                          KC.NO,   KC.NO,                                                         KC.P0, KC.PDOT,
+                                                 KC.TRNS, KC.TRNS,       KC.TRNS,   KC.NO,
+                                                 KC.TRNS, KC.TRNS,       KC.TRNS, KC.TRNS,
+                                                 KC.TRNS, KC.TRNS,       KC.TRNS, KC.TRNS
+    ],
+    [  #2
+        KC.F13,  KC.F14,   KC.F15,  KC.F16,  KC.F17, KC.F18,                         KC.F19,    KC.20, KC.21,   KC.22,   KC.23,   KC.24,
```
Comment (xs5871) [inline boards/dactyl_manuform/5x6/main.py:43]: `KC.20` - `KC.24` aren't valid key codes, you probably forgot an "F" in there. This is a copy&paste mistake that appears in other keymaps here as well.

File: boards/dactyl_manuform/readme.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+*Two wired Dactyl Manuform 4x6 variants*
+
+## Variants
+
+Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  
+
+As standard: 
+- The finger keywell bottom row has 2 keys, 1 each in ring and middle columns
+    - Exception to this rule is the 7 column variants, that have two additional keys in this row
+- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:15]:
```suggestion
V ariants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:
- The finger key-well bottom row has 2 keys, 1 each in ring and middle columns. Exception to this rule is the 7 column variants that have two additional keys in this row.
- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows .
```

Is "keywell" the proper spelling? It looks like a compositum that should have a hyphen, but I'm not a linguist.

File: boards/dactyl_manuform/readme.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The [Dactyl Manuform](https://github.com/tshort/dactyl-keyboard) is a handwired, split bodied, tented, curved keywell keyboard - adapted from the [Dactyl](https://github.com/adereth/dactyl-keyboard) keyboard with the thumb cluster design of the [ManuForm](https://geekhack.org/index.php?topic=46015.0) keyboard. Information for building a Dactyl Manuform found in first link.
+
+![Imgur](https://i.imgur.com/7y0Vbyd.jpg)
+*Two wired Dactyl Manuform 4x6 variants*
+
+## Variants
+
+Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  
+
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:11]:
```suggestion
```

File: boards/dactyl_manuform/readme.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Dactyl Manuform
+
+The [Dactyl Manuform](https://github.com/tshort/dactyl-keyboard) is a handwired, split bodied, tented, curved keywell keyboard - adapted from the [Dactyl](https://github.com/adereth/dactyl-keyboard) keyboard with the thumb cluster design of the [ManuForm](https://geekhack.org/index.php?topic=46015.0) keyboard. Information for building a Dactyl Manuform found in first link.
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:3]: Break up long lines to preferable 80 characters.
External links should have the current date attached (i.e.: last known valid date), if there's the potential that the link may change withing a couple of years. (Yes, that caused issues in the past3)

File: boards/dactyl_manuform/readme.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## Variants
+
+Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  
+
+As standard: 
+- The finger keywell bottom row has 2 keys, 1 each in ring and middle columns
+    - Exception to this rule is the 7 column variants, that have two additional keys in this row
+- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows 
+
+Variants are denoted as `RowCount`*x*`ColumnCount`
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:17]:
```suggestion
```

File: boards/dactyl_manuform/readme.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+### Row
+| Count | Description |
+| --- | --- |
+| 4 | Three rows, typically for alphabet and some puncuation characters, with 2 key (finger keywell) bottom row |
+| 5 | As *4 row* with number row above |
+| 6 | As *5 row* with function row above |
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:24]:
```suggestion
| 4 | Three rows, typically for alphabet and some punctuation characters, with 2 key (finger keywell) bottom row |
| 5 | Like *4 rows* with number row above |
| 6 | Like *5 rows* with function row above |
```

File: boards/dactyl_manuform/readme.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+### Column
+| Count | Description |
+| --- | --- |
+| 5 | A column for each finger with additional column for first finger |  
+| 6 | As *5 column* with additional pinky finger column |
+| 7 | As *6 column* with either an additional first finger column (`5x7`) or additional pinky column (`6x7`) |
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:31]:
```suggestion
| 5 | A column for each finger with additiona index finger column |  
| 6 | Like *5 columns* with additional pinky finger column |
| 7 | Like *6 columns* with either an additional index finger column (`5x7`) or additional pinky column (`6x7`) |
```

File: boards/dactyl_manuform/readme.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+## Microcontroller support
+
+Amend `controller` in the following line of desired variant's `kb.py` file to supported microcontroller listed in `kmk/quickpin/pro_micro`:
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:35]: If you have to edit it, what's the point of having 6 different `kb.py`s in the first place? Why not a single one for 4x5 with the additional rows/colums commented out?
Edit: I guess the `coord_mapping` would be an issue, althogh most of that could be alleviated by reordering the column pins. New request: reorder the column pins to make the `coord_map` as linear as possible.

File: boards/dactyl_manuform/readme.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+| 6 | As *5 row* with function row above |
+
+### Column
+| Count | Description |
+| --- | --- |
+| 5 | A column for each finger with additional column for first finger |  
+| 6 | As *5 column* with additional pinky finger column |
+| 7 | As *6 column* with either an additional first finger column (`5x7`) or additional pinky column (`6x7`) |
+
+## Microcontroller support
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:33]:
```suggestion
## Microcontroller Support
```

File: boards/dactyl_manuform/readme.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+For example, nice!nano controller(s): 
+```python
+from kmk.quickpin.pro_micro.nice_nano import pinout as pins
+```
+
+## KMK Specifics
+
+Extentions enabled by default:  
+- [Layers](/docs/en/layers.md): As many as you want/need
+- [Split](/docs/en/split_keyboards.md): Configured to 1 wire UART to match legacy configuration. Please see documentation for enabling 2 wire UART or, for capable controllers, Bluetooth
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:49]:
```suggestion
- [Split](/docs/en/split_keyboards.md): Configured to 1-wire UART to match legacy configuration. Please see documentation for enabling 2-wire UART or, for capable controllers, Bluetooth.
```

Again, split long lines.

**Review by xs5871 — CHANGES_REQUESTED** [review]
Here's the first batch of contentwise change requests.
There's going to be at least a second round to make the linter happy.

File: boards/dactyl_manuform/readme.md (hunk 8 above)
Comment (dunk2k) [reply boards/dactyl_manuform/readme.md:31]: Sorry @xs5871 , I have no idea what entails.

My understanding is the `coord_mapping` is directly related to pin matrix, thus there is a different coord_mapping per keyboard as their pin matrices are unique.

Would you happen to have a reference to your request?

File: boards/dactyl_manuform/5x6/main.py (hunk 1 above)
Comment (dunk2k) [reply boards/dactyl_manuform/5x6/main.py]: Corrected in commit `bd94164`

File: boards/dactyl_manuform/readme.md (hunk 4 above)
Comment (dunk2k) [reply boards/dactyl_manuform/readme.md:3]: Hi @xs5871,

looking at other readme.md files I'm not seeing a date attached to external links. Could you give example of how date should be added?

Comment (xs5871) [reply boards/dactyl_manuform/readme.md:3]: Yeah, sadly we never finished the doc linting and doc guidlines.
"[text (YYYY-MM)](link)" should do. Not strictly necessary for expected long-lived external links; github should be fine without, unless it's a small personal project. I wasn't familiar with geekhack, but they seem to be very long lived as well, although that could benefit from a date of post creation.
The imgur could state roughly when the image was taken, for example.

File: boards/dactyl_manuform/readme.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+The [Dactyl Manuform](https://github.com/tshort/dactyl-keyboard) is a handwired, split bodied, tented, curved keywell keyboard - adapted from the [Dactyl](https://github.com/adereth/dactyl-keyboard) keyboard with the thumb cluster design of the [ManuForm](https://geekhack.org/index.php?topic=46015.0) keyboard. Information for building a Dactyl Manuform found in first link.
+
+![Imgur](https://i.imgur.com/7y0Vbyd.jpg)
+*Two wired Dactyl Manuform 4x6 variants*
+
+## Variants
+
+Dactyl Manuform's are built in variations that cater for different row and column counts, and thumb clusters.  
+V ariants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:11]:
```suggestion
Variants are denoted as `RowCount`*x*`ColumnCount` and share the common configuration of:
```

File: boards/dactyl_manuform/readme.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+- The finger key-well bottom row has 2 keys, 1 each in ring and middle columns. Exception to this rule is the 7 column variants that have two additional keys in this row.
+- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows .
```
Comment (xs5871) [inline boards/dactyl_manuform/readme.md:13]:
```suggestion
- The thumb cluster has 6 keys, arranged in 2 columns by 3 rows.
```

File: boards/dactyl_manuform/readme.md (hunk 8 above)
Comment (xs5871) [reply boards/dactyl_manuform/readme.md:31]: No, sorry, strike that. Didn't consider that these are all split boards. Let's leave it the way it is.

**Review by xs5871 — COMMENTED** [review]
Please verify your next commit with `make test` before submitting.

File: boards/dactyl_manuform/readme.md (hunk 4 above)
Comment (dunk2k) [reply boards/dactyl_manuform/readme.md]: date added for ManuForm link

**Review by xs5871 — COMMENTED** [review]
General request: Please don't push every single commit individually to the PR. I've been away for a couple of days and have 70 notifications about incremental fixes; it got spammy after about 5.
Additionally, if you know that a PR is "work in progress", mark it as a draft PR and only promote it to a reviewable PR once you're done. That way I don't have to guess if it's ready or not.

**Comment by dunk2k** [comment]
> lease don't push every single commit individually to the PR. I've been away for a couple of days and have 70 notifications about incremental fixes; it got spammy after about 5.
Additionally, if you know that a PR is "work in progress", mark it as a draft PR and only promote it to a reviewable PR once you're done

duly noted

**Review by xs5871 — APPROVED** [review]
Don't forget to run `make fix-isort fix-formatting test` on your other PRs.

**Pull request merged**